test/boost/sstable_compaction_test.cc

```diff
@@ -2120,12 +2120,12 @@ SEASTAR_THREAD_TEST_CASE(sstable_scrub_validate_mode_test) {
         BOOST_REQUIRE(sstables.size() == 1);
         auto sst = sstables.front();
 
-        // No way to really test validation besides observing the log messages.
         sstables::compaction_type_options::scrub opts = {
             .operation_mode = sstables::compaction_type_options::scrub::mode::validate,
         };
-        table->get_compaction_manager().perform_sstable_scrub(ts, opts, tasks::task_info{}).get();
+        auto stats = table->get_compaction_manager().perform_sstable_scrub(ts, opts, tasks::task_info{}).get();
 
+        BOOST_REQUIRE(stats.has_value() && stats->validation_errors > 0);
```

Comment:
```
BOOST_REQUIRE(stats.has_value());
BOOST_REQUIRE_EQUAL(stats->validation_errors, 0);
```
This will give much better diagnostics if the test fails. You will know exactly which part of the above expression failed and how. With the combined check, all you know is that the check failed, without any idea about why.